tasks/managed/extract-binaries-from-image/extract-binaries-from-image.yaml

```
@@ -55,8 +55,8 @@ spec:
         DESIRED_COMPONENTS_LIST=
         if [ ! -f "${DATA_FILE}" ] ; then
             echo "No data JSON was provided."
-        elif [ "$(jq '."contentGateway" | has("components")' "${DATA_FILE}")" = true ]; then
-            DESIRED_COMPONENTS_LIST="$(jq -r '."contentGateway".components[].name' "${DATA_FILE}")"
+        elif [ "$(jq 'try .mapping.components[0].contentGateway | has("components")' "${DATA_FILE}")" = true ]; then
```

What is the reason for the `try`? Just curious. Even if I have an empty json, it still succeeds for me without it:
```
jq '.mapping.components[0].contentGateway | has("components")' <<< '{}'
false
```

Even if I try a few more scenarios:
```
jq '.mapping.components[0].contentGateway | has("components")' <<< '{"mapping":{"components": []}}'
false
jq '.mapping.components[0].contentGateway | has("components")' <<< '{"mapping":{"components": [{}]}}'
false
```

---

I added it to check for the case where a user might define components as objects instead of arrays, but thinking about it now, it should error, and it should already be caught at check-data-keys. 

When I was changing over the RPA I was doing this first .... 
```
{
  "mapping": {
    "components": { 
      "contentGateway": {
        "components": [],
      }
    }
  }
}
```

With try:
```
jq 'try .mapping.components[0].contentGateway | has("components")' <<< '{"mapping":{"components":{}}}'
```
Without try:
```
jq ' .mapping.components[0].contentGateway | has("components")' <<< '{"mapping":{"components":{}}}'
jq: error (at <stdin>:1): Cannot index object with number
```

---

OK, I see. So it's probably not needed. But if you want to keep it, it's fine with me. Up to you.

---

I am going to remove it as it would silently skip  `DESIRED_COMPONENTS` if they were not in the correct format, and the user wouldn't know why it didn't work correctly.

---

I see this is removed in the new PR: https://github.com/konflux-ci/release-service-catalog/pull/975/files